src/preamble.js

```diff
@@ -295,8 +295,20 @@ function updateGlobalBufferAndViews(buf) {
   Module['HEAPF32'] = HEAPF32 = new Float32Array(buf);
   Module['HEAPF64'] = HEAPF64 = new Float64Array(buf);
 #if WASM_BIGINT
-  Module['HEAP64'] = HEAP64 = new BigInt64Array(buf);
-  Module['HEAPU64'] = HEAPU64 = new BigUint64Array(buf);
+  if (typeof BigInt64Array !== "undefined") {
```

Comment:
I think this check should go inside the #if MIN_SAFARI_VERSION < 150000 check.   We don't want all uses to pay the price of checking against undefined here.